classes/curlmanager_security_helper.php

```diff
@@ -22,150 +29,134 @@
  * @copyright Catalyst IT
  * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
  */
-namespace tool_curlmanager;
-
-use core\files\curl_security_helper_base;
-use core\files\curl_security_helper;
-
 class curlmanager_security_helper extends curl_security_helper_base {
     /**
      * url_is_blocked.
      *
      * @param string $urlstring the URL to check.
      * @return bool true if the URL is blocked or false if the URL is allowed.
      */
-    public function url_is_blocked($urlstring) : bool {
+    public function url_is_blocked($urlstring): bool {
+        try {
+            $curlsecurityhelper = new curl_security_helper();
+            $blockedbymoodle = $curlsecurityhelper->url_is_blocked($urlstring);
+
+            $url = new \moodle_url($urlstring);
+            $blockedbyus = !$this->host_is_whitelisted($url->get_host());
+
+            // Log the result.
+            $this->log_curl_http_requests($url, $blockedbymoodle, $blockedbyus);
 
-        // Log the http request.
-        return $this->log_curl_http_requests($urlstring);
+            return $blockedbymoodle || $blockedbyus;
+        } catch (Throwable $e) {
+            // Block if there is any error.
+            return true;
```

Comment:
I think failing open here is the safer call. In conjunction with password validator, this may prevent logins at all